### PR TITLE
Add  gethrtime() and dummy getrlimit() setrlimit(), add some dependency

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary
Add  gethrtime() and dummy getrlimit() setrlimit(), add some dependency

### CODE REVIEW

**Comment by v01d** [comment]
Could you please also add documentation for `gethrtime` in "clocks & timers" user reference page of the `Documentation`?

**Review by v01d — CHANGES_REQUESTED** [review]
Add documentation for gethrtime

**Comment by GUIDINGLI** [comment]
@v01d done

File: Documentation/conf.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
 # -- Project information -----------------------------------------------------
 
+master_doc = 'index'
```
Comment (v01d) [inline Documentation/conf.py:40]: why is this being done here?

File: libs/libc/unistd/lib_getrlimit.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+int getrlimit(int resource, FAR struct rlimit *rlp)
+{
+  if (rlp == NULL)
+    {
+      set_errno(EINVAL);
+      return ERROR;
+    }
+
+  memset(rlp, 0, sizeof(*rlp));
+  return OK;
```
Comment (v01d) [inline libs/libc/unistd/lib_getrlimit.c:54]: can you add a comment to indicate this is a stub?

File: libs/libc/unistd/lib_setrlimit.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * Description:
+ *   The getrlimit() and setrlimit() system calls get and
+ *   set resource limits respectively.
+ *
+ ****************************************************************************/
+
+int setrlimit(int resource, FAR const struct rlimit *rlp)
+{
+  return OK;
```
Comment (v01d) [inline libs/libc/unistd/lib_setrlimit.c:47]: same as the other function

File: Documentation/conf.py (hunk 1 above)
Comment (GUIDINGLI) [reply Documentation/conf.py:40]: https://stackoverflow.com/questions/56336234/build-fail-sphinx-error-contents-rst-not-found

File: libs/libc/unistd/lib_getrlimit.c (hunk 2 above)
Comment (GUIDINGLI) [reply libs/libc/unistd/lib_getrlimit.c]: done

File: libs/libc/unistd/lib_setrlimit.c (hunk 3 above)
Comment (GUIDINGLI) [reply libs/libc/unistd/lib_setrlimit.c:49]: done

File: Documentation/conf.py (hunk 1 above)
Comment (v01d) [reply Documentation/conf.py:40]: While this is the readthedocs theme it is not the same as hosting on readthedocs.org.
I have not encountered any issue and CI build documentation correctly. Can you copy the error you are receiving?
Also, please verify you are using the expected versions of all sphinx packages. The safest why is doing it like it is described here: https://nuttx.apache.org/docs/latest/contributing/documentation.html

**Comment by v01d** [comment]
You will have to rebase since we merged something that conflicts with this.

File: Documentation/conf.py (hunk 1 above)
Comment (GUIDINGLI) [reply Documentation/conf.py:40]: Sphinx error:
master file /home/ligd/platform/mirtos3/nuttx/Documentation/contents.rst not found
Makefile:42: recipe for target 'html' failed
make: *** [html] Error 2

Maybe should use the install cmds as your recommended, but mine setup the sphinx as follows:
sudo  pip install m2r2
sudo apt-get install python-pip
pip install -U sphinx
sudo pip install sphinx_rtd_theme

This is caused by the conflicts of the default sphinx version in readthedocs and the local environment.
See:
https://github.com/readthedocs/readthedocs.org/issues/2569

I think we should cover this situation.

Comment (v01d) [reply Documentation/conf.py:40]: But again, we're not using that service. We're simply using their theme.
The point is that we have a working set of package versions for all these dependencies so that we agree on a working version. Please follow the approach in the documentation and see if you still get the error.

Comment (GUIDINGLI) [reply Documentation/conf.py]: sphinx-build 1.8.5 will meet compile error:
master file /home/ligd/platform/mirtos3/nuttx/Documentation/contents.rst not found

after update to sphinx-build 3.3.0+:
then compile pass

remove the modify as you like.

Comment (v01d) [reply Documentation/conf.py]: Great, thanks

**Review by v01d — DISMISSED** [review]

**Comment by v01d** [comment]
Please have a look at the CI failure

**Comment by GUIDINGLI** [comment]
@v01d done

File: include/termios.h [hunk 4]
```diff
@@ -248,6 +250,13 @@ struct termios
   tcflag_t  c_cflag;        /* Control modes */
   tcflag_t  c_lflag;        /* Local modes */
   cc_t      c_cc[NCCS];     /* Control chars */
+
+  /* Implementation specific fields.  For portability reasons, these fields
+   * should not be accessed directly, but rather through only through the
+   * cf[set|get][o|i]speed() POSIX interfaces.
+   */
+
+  speed_t c_speed;          /* Input/output speed (non-POSIX) */
```
Comment (xiaoxiang781216) [inline include/termios.h]: Revert the change, which is removed by PR https://github.com/apache/incubator-nuttx/pull/1767

File: include/cxx/cunistd [hunk 5]
```diff
@@ -71,7 +71,9 @@ namespace std
 
   // Terminal I/O
 
+#ifdef CONFIG_SERIAL_TERMIOS
```
Comment (xiaoxiang781216) [inline include/cxx/cunistd:74]: let's move this change to patch "include/termios.h: undefine symbols when !CONFIG_SERIAL_TERMIOS"

**Review by xiaoxiang781216 — CHANGES_REQUESTED** [review]

File: include/termios.h (hunk 4 above)
Comment (GUIDINGLI) [reply include/termios.h]: done

File: include/cxx/cunistd (hunk 5 above)
Comment (GUIDINGLI) [reply include/cxx/cunistd:74]: done

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM. @v01d please merge it if you don't more comment.

File: include/cxx/cunistd (hunk 5 above)
Comment (v01d) [reply include/cxx/cunistd:74]: I still see the change there

Comment (xiaoxiang781216) [reply include/cxx/cunistd:74]: The change is required, @GUIDINGLI already merge the change related CONFIG_SERIAL_TERMIOS into one patch.

Comment (v01d) [reply include/cxx/cunistd:74]: Ah, I thought I wanted to have this in a separate PR. Ok, I'll merge then
